lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,411 @@
+    '''
+    # Decouple local variables from variables supplied
+    # Ensure that variables are in correct units
+    temp = temperature.copy()
+    temp.convert_units('celsius')
+    rh = relative_humidity.copy()
+    rh.convert_units(1)
+    press = pressure.copy()
+    press.convert_units('kPa')
+    svp = saturation_vapour_pressure_goff_gratch(temp, press)
+    svp.convert_units('kPa')
+
+    # Calculation
+    result_numer = (0.62198 * rh.data * svp.data)
+    result_denom = (press.data - (rh.data * svp.data))
+    hr = temp.copy(data=result_numer / result_denom)
+
+    # Tidying up cube
+    hr.rename("Humidity ratio")
+    hr.units = Unit("1")
+    return hr
+
+
+def humidity_ratio_fm_wb(temperature, wet_bulb, pressure):
+    ''' Function to compute humidity ratio given a temperature pressure and
+    wet bulb temperature.
+
+    Args:
+        temp: iris.cube.Cube
+            Cube of temperature which will be converted to Celsius
+            prior to calculation
+
+        wet_bulb: iris.cube.Cube
+            Cube of wet_bulb_temperature which will be converted to Celsius
+            prior to calculation
+
+        pressure: iris.cube.Cube
+            Cube of pressure which will be converted to kiloPascals
+            prior to calculation
+
+    Returns:
+        humidity_ratio : iris.cube.Cube
+            humidity ratio cube with units(1)
+
+    References:
+        ASHRAE Fundamentals handbook (2005) Equation 22, 24, p6.8
+    '''
+    temp = temperature.copy()
+    temp.convert_units("celsius")
+    press = pressure.copy()
+    press.convert_units("kPa")
+    wb = wet_bulb.copy()
+    wb.convert_units("celsius")
+
+    # Calculation
+    wb_depression = temp - wb
+    svp = saturation_vapour_pressure_goff_gratch(temp, press)
+    svp.convert_units('kPa')
+    ws = 0.62198 * svp / (press - svp)
+    # This is inefficient because it produces both answers
+    above_zero = (((2501. - 2.326 * wb.data) * ws.data -
+                   1.006 * wb_depression.data) /
+                  (2501. + 1.86 * temp.data - 4.186 * wb.data))
+    below_zero = (((2830. - 0.24 * wb.data) * ws.data -
+                   1.006 * wb_depression.data) /
+                  (2830. + 1.86 * temp.data - 2.1 * wb.data))
+    result = np.where(temp.data >= 0, above_zero, below_zero)
+    hr = temp.copy(data=result)
+
+    # Tidy up
+    hr.units = Unit(1)
+    hr.rename("Humidity ratio")
+    return hr
+
+
+def wet_bulb(temperature, relative_humidity, pressure,
+             p_units="hPa", precision=0.00001):
+    """
+    Calculates the Wet Bulb Temperature Using Newton-Raphson iteration
+
+    Args:
+        temperature : iris.cube.Cube
+            Cube of temperature which will be converted to Celsius
+            prior to calculation
+        rel_humidity : iris.cube.Cube
+            Cube of relative humidity
+        pressure : iris.cube.Cube or float
+            Cube of pressure which will be converted to kilopPascals
```

Comment:
Typo: KiloPascals

Reply:
The typo was the additional "p", KilopPascals, I have no problem with your capitalisation.